packages/core/src/client.ts

```diff
@@ -2227,12 +2227,36 @@ export class MedplumClient extends EventTarget {
   }
 }
 
+/**
+ * Returns the current window if available.
+ * All access to the current window should use this to support SSR such as Next.js.
+ * @returns The current window or undefined if not available.
+ */
+function getWindow(): Window | undefined {
+  return typeof window === 'undefined' ? undefined : window;
+}
+
+/**
+ * Returns the default fetch method.
+ * The default fetch is currently only available in browser environments.
+ * If you want to use SSR such as Next.js, you should pass a custom fetch function.
+ * @returns The default fetch function for the current environment.
+ */
+function getDefaultFetch(): FetchLike {
+  const window = getWindow();
+  if (!window) {
+    throw new Error('Fetch not available in this environment');
+  }
+  return window.fetch.bind(window);
+}
+
 /**
  * Returns the base URL for the current page.
  * @category HTTP
  */
 function getBaseUrl(): string {
-  return window.location.protocol + '//' + window.location.host + '/';
+  const window = getWindow();
```

Comment:
I know this isn't actually the point of this PR, but why doesn't this return `this.#baseURL`. The naming is confusing about whether the "BaseURL" is the baseURL for the FHIR server, or the baseURL of the current app

Reply:
Yeah... that's a fair critique.... this is ancient code.  It's particularly bad because there are two `getBaseUrl` methods in the same file that do very different things.  I'll rename.